src/citationnet/static/force-citationNet.js

```diff
@@ -0,0 +1,419 @@
+import * as THREE from './three.module.js';
+import * as Lut from './lut.js';
+
+/**
+* Async function for fetching and parsing JSON files
+* @param {String} path - path or url to JSON file
+* @returns {object} parsed JSON object
+*/
+async function fetchJSON(data) {
+    // console.log('parsed', data)
+
+    try {
+        // waits until the request completes...
+        const retdata = await JSON.parse(data);
+    } catch (error) {
+        console.log(data);
+        throw error;
+    }
+    return retdata;
+}
+
+
+class CitationNet {
+
+    /**
+    * Constructs a new CitationNet object, but does not initialize it. Call object.initialize() right after this.
+    * @param {String} jsondata - path or url to citation data as JSON file/stream
+    */
+    constructor(jsondata = null) {
+        // if jsonPath is not provided try to get globJsonPath, show alert if fails
+        try {
+            if (!(jsondata)) jsondata = jsondata;
+        } catch (error) {
+            alert("no JSON containing citation data specified, graph cannot be displayed");
```

Comment:
I strongly recommend not to use `alert()`. It blocks the browser and the javascript thread, forcing the user to interact to continue doing anything else at all. Some popup blockers may block it anyway. Instead, consider showing a UI component with this information. Since you're using Bootstrap, consider using an alert or a toast. You can generate the HTML dynamically and append it to the DOM (and remove it when dismissed or after a timeout). If that sounds too complicated, you may also consider setting a property to the class and have the code in `visDynamic.html` monitor the property to react accordingly (e.g. by showing some text in the main `<div>`...)